metomi/rose/date_cli.py

```diff
@@ -177,6 +177,40 @@
 
 from metomi.isodatetime.main import main as iso_main
 
+from metomi.rose.date import LEGACY_OFFSET, upgrade_offset
+
+
+def _handle_old_offsets(args: list) -> list:
+    """Handle Legacy Rose date --offset values:
+
+    # https://github.com/metomi/rose/issues/2577
+
+
+    Examples:
+    >>> _handle_old_offsets(['rose-date', '--offset=1d1s'])
+    [WARN] This offset syntax is deprecated.
+    ['rose-date', '--offset=P1DT0H0M1S']
+    >>> _handle_old_offsets(['rose-date', '-s', '1d1s'])
+    [WARN] This offset syntax is deprecated.
+    ['rose-date', '-s', 'P1DT0H0M1S']
+    """
+    for index in range(len(args)):
+        arg = args[index]
+        if '--offset' in arg or '-s' in arg:
+            # Case: --offset=<offset> is a single item in args list:
+            if (
+                len(arg.split('=')) > 1
```

Comment:
Absolutely fine but can be done more simply:

```suggestion
                '=' in arg
```